x-pack/legacy/plugins/canvas/public/angular/controllers/canvas.tsx

```diff
@@ -8,44 +8,47 @@ import { render, unmountComponentAtNode } from 'react-dom';
 import { I18nProvider } from '@kbn/i18n/react';
 import { Provider } from 'react-redux';
 import { Store } from 'redux';
-import chrome from 'ui/chrome';
-import { UICapabilities } from 'ui/capabilities';
+import { CanvasStartDeps, CoreStart } from '../../plugin';
+import { KibanaContextProvider } from '../../../../../../../src/plugins/kibana_react/public';
 
 // @ts-ignore Untyped local
 import { App } from '../../components/app';
 import { AngularStrings } from '../../../i18n';
 
 const { CanvasRootController: strings } = AngularStrings;
 
-export function CanvasRootController(
-  canvasStore: Store,
-  $scope: any, // Untyped in Kibana
-  $element: any, // Untyped in Kibana
-  uiCapabilities: UICapabilities
-) {
-  const domNode = $element[0];
+export function CanvasRootControllerFactory(coreStart: CoreStart, plugins: CanvasStartDeps) {
+  return function CanvasRootController(
+    canvasStore: Store,
```

Comment:
seems like this could just be de-coupled from angular and instantiated in the top-level plugin class